pkgs/build-support/ocaml/dune.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchurl, ocaml, findlib, dune }:
+
+{ pname, version, src, buildInputs ? [], meta ? {}, ... }@args:
+
+if args ? minimumOCamlVersion &&
+   ! stdenv.lib.versionAtLeast ocaml.version args.minimumOCamlVersion
+then throw "${pname} is not available for OCaml ${ocaml.version}"
+else
+
+stdenv.mkDerivation (args //
+
+(if src ? dune-release && src.dune-release then
+{
+  unpackCmd = "tar -xjf $curSrc";
+}
+else { }) //
+
+(if src ? github-owner then
+let
+ github-repo-url =
+  "https://github.com/${src.github-owner}/${if src ? repo then src.repo else pname}";
+
+ rev = if src ? rev then src.rev else version;
+in
+{
+  src = fetchurl {
+    url =
+      if src ? dune-release && src.dune-release then
+        "${github-repo-url}/releases/download/${rev}/${pname}-${version}.tbz"
```

Comment:
Do they depend on github as a single source of truths?

Reply:
This is when both dune-release and github-owner are defined. You can also specify dune-release with another src.